tests/test_land_vision.py

```diff
@@ -19,3 +25,36 @@
 #     print ('ret', json.dumps(ret['tile'], indent = 2))
 
 #     # _land_vision.seeLand(imageUrl)
+
+def test_urban_tree_detection_inference():
+    img_path = "./uploads/images/naip_image.png"
+    weights_dir = "./land_vision/urban_tree_detection/pretrained"
+    min_distance = 1
+    threshold_abs = None
+    threshold_rel = 0.2
+    max_distance = 10
+
+    img = cv2.imread(img_path)
+    img_rgb = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)  
+    images = img_rgb[None,:,:,:]
+    
+    preprocess = eval(f'preprocess_RGB')
+    training_model, model = SFANet.build_model(
+        images.shape[1:],
+        preprocess_fn=preprocess)
+
+    weights_path = os.path.join(weights_dir,'weights.best.h5')
+    training_model.load_weights(weights_path)
+
+    preds = model.predict(images,verbose=True,batch_size=1)[...,0]
```

Comment:
For adding to vector tiles database, is this the main function? So we'd pass in the image tile and get the tree lng lats back and store those? If so I can help with that.
In general, you'd create a polygon object, e.g. https://github.com/teal-towns/tealtowns/blob/main/mapbox/mapbox_vector_tile.py#L29
https://github.com/teal-towns/tealtowns/blob/main/db_schema_vector_tiles.json#L56
And then save that polygon in the database for the appropriate tile (this function will work, or we can make a single get tile version if that is easier https://github.com/teal-towns/tealtowns/blob/main/vector_tiles/vector_tiles_data.py#L31 ). Let me know if you'd like me to code it up, or if you want to try first.

Reply:
this function predicts a confidence map I think, the main function is  the one right below it, results = predict(preds=preds,...). results is a dictionary, the tree_locs are stored in results['pred_locs']. 
I think it is better that you code it cause I don't understand this func GetTiles well, I don't see anywhere to put the created polygon obj.

Reply:
Ok, I'll add in the code, probably on Tuesday.